loc-analysis.exs

```diff
@@ -0,0 +1,202 @@
+#! /usr/bin/env elixir
+
+Mix.install([
+  {:csv, "~> 3.0"}
+])
+
+defmodule GitCommandError do
+  defexception message: "Git command failed."
+
+  @impl true
+  def exception(value) do
+    case value do
+      [] -> %GitCommandError{}
+      _ -> %GitCommandError{message: "Git command failed: " <> value}
+    end
+  end
+end
+
+defmodule Git do
+  def get_snapshot!(commit_hash) do
+    with destination_path <- System.tmp_dir!(),
+         {:ok, tar_file} <- call_git_archive!(commit_hash, destination_path),
+         :ok <- call_tar_extract!(tar_file, destination_path),
+         :ok <- Path.join(destination_path, tar_file) |> File.rm!() do
+      {:ok, Path.join(destination_path, commit_hash)}
+    end
+  end
+
+  def monthly_commits do
+    call_git_log!()
+    |> String.split("\n")
+    |> Stream.uniq_by(&year_month_of/1)
+  end
+
+  def weekly_commits do
```

Comment:
Random idea (since you mentioned that you haven't used Elixir for a long time): have you considered writing [Doctests](https://elixir-lang.org/getting-started/mix-otp/docs-tests-and-with.html#doctests)? I love them because you document real examples and you can actually run them.